src/formatDate.js

```diff
@@ -50,7 +50,124 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
```

Comment:
Too much of code are repeated in few cases, that's not how this task should be done. Refactor your code and get rid of all repeated code parts